### PR TITLE
Implements RgaTreeSplit

### PR DESCRIPTION
**Pull request by 7hong13**
#### What this PR does / why we need it?
Implements `RgaTreeSplit` based on `JS-SDK` version.

#### Any background context you want to provide?
I didn’t put `RgaTreeSplitNode`, `RgaTreeSplitNodeID`, and `RgaTreeSplitNodePos` inside `RgaTreeSplit` since they are also used from outside.

I changed all the `editedAt` parameter name to `executedAt` based on [the previous discussion](https://github.com/yorkie-team/yorkie-ios-sdk/pull/13#discussion_r987486485), but maybe using `editedAt` might convey the meaning better in this context. 

I used [TreeMap](https://docs.oracle.com/javase/7/docs/api/java/util/TreeMap.html) for LLRB.

#### What are the relevant tickets?

Fixes #

### Checklist

- [x] Added relevant tests or not required
- [x] Didn't break anything

### CODE REVIEW

File: yorkie/src/main/kotlin/dev/yorkie/document/crdt/RgaTreeSplit.kt [hunk 1]
```diff
@@ -0,0 +1,575 @@
...
+            }
+            caretPos = RgaTreeSplitNodePos(inserted.id, inserted.contentLength)
+        }
+
+        // 4. Add removed nodes.
+        removedNodeMapByNodeKey.forEach {
+            removedNodeMap[it.key] = it.value
+        }
+
+        return Triple(caretPos, latestCreatedAtMap, changes)
```
Comment (skhugh) [inline yorkie/src/main/kotlin/dev/yorkie/document/crdt/RgaTreeSplit.kt:99]: added TextChange element would be lost as we are passing the instance created by `toMutableList`

**Review by skhugh — APPROVED** [review]

**Review by hackerwins — APPROVED** [review]
Thanks for your contribution.

File: yorkie/src/main/kotlin/dev/yorkie/document/crdt/RgaTreeSplit.kt [hunk 2]
```diff
@@ -0,0 +1,577 @@
...
+    /**
+     * Clears the index nodes of the given deletion [boundaries].
+     * The [boundaries] mean the nodes that will not be deleted in the range.
+     */
+    private fun deleteIndexNodes(boundaries: List<RgaTreeSplitNode<T>?>) {
+        for (index in 0..boundaries.size - 2) {
+            val leftBoundary = boundaries[index]
+            val rightBoundary = boundaries[index + 1]
+            // NOTE(7hong13): below comment has a typo in JS-SDK.
+            // If there is no node to delete between boundaries, do nothing.
```
Comment (7hong13) [inline yorkie/src/main/kotlin/dev/yorkie/document/crdt/RgaTreeSplit.kt:332]: @hackerwins I left some notes on the comments I fixed.
(refer to: [link](https://github.com/yorkie-team/yorkie-js-sdk/blob/893b2c9a870129068d152e092b2908b53fef8e72/src/document/crdt/rga_tree_split.ts#L940))

File: yorkie/src/main/kotlin/dev/yorkie/document/crdt/RgaTreeSplit.kt [hunk 3]
```diff
@@ -0,0 +1,577 @@
...
+            if (leftBoundary?.next != rightBoundary) {
+                treeByIndex.deleteRange(requireNotNull(leftBoundary), rightBoundary)
+            }
+        }
+    }
+
+    // NOTE(7hong13): original comment from JS-SDK:
+    // `findNodePos` finds RGATreeSplitNodePos of given offset.
+    /**
+     * Finds [RgaTreeSplitNodePos] of the given [index].
```
Comment (7hong13) [inline yorkie/src/main/kotlin/dev/yorkie/document/crdt/RgaTreeSplit.kt:342]: @hackerwins I left some notes on the comments I fixed.
(refer to: [link](https://github.com/yorkie-team/yorkie-js-sdk/blob/893b2c9a870129068d152e092b2908b53fef8e72/src/document/crdt/rga_tree_split.ts#L541))

File: yorkie/src/main/kotlin/dev/yorkie/document/crdt/RgaTreeSplit.kt [hunk 4]
```diff
@@ -0,0 +1,577 @@
...
+    private fun splitValue(offset: Int): T {
+        val valueBefore = _value
+        _value = valueBefore.subSequence(0, offset) as T
+        return valueBefore.subSequence(offset, valueBefore.length) as T
+    }
+
+    // NOTE(7hong13): original comment from JS-SDK:
+    // `canDelete` checks if node is able to delete.
+    /**
+     * Checks if this [RgaTreeSplitNode] can be deleted or not.
```
Comment (7hong13) [inline yorkie/src/main/kotlin/dev/yorkie/document/crdt/RgaTreeSplit.kt:521]: @hackerwins I left some notes on the comments I fixed.
(refer to: [link](https://github.com/yorkie-team/yorkie-js-sdk/blob/893b2c9a870129068d152e092b2908b53fef8e72/src/document/crdt/rga_tree_split.ts#L396))

File: yorkie/src/main/kotlin/dev/yorkie/document/crdt/RgaTreeSplit.kt [hunk 5]
```diff
@@ -0,0 +1,577 @@
...
+     */
+    fun canDelete(executedAt: TimeTicket, latestCreatedAt: TimeTicket): Boolean {
+        return createdAt <= latestCreatedAt && (isRemoved || _removedAt < executedAt)
+    }
+
+    // NOTE(7hong13): original comment from JS-SDK:
+    // `remove` removes node of given edited time.
+    /**
+     * Removes this [RgaTreeSplitNode] at the given [executedAt].
```
Comment (7hong13) [inline yorkie/src/main/kotlin/dev/yorkie/document/crdt/RgaTreeSplit.kt:530]: @hackerwins I left some notes on the comments I fixed.
(refer to: [link](https://github.com/yorkie-team/yorkie-js-sdk/blob/893b2c9a870129068d152e092b2908b53fef8e72/src/document/crdt/rga_tree_split.ts#L406))

I will remove all the notes in `RgaTreeSplit` in next PR.

**Pull request merged**